diff -uNr LuaJIT-riscv/src/lj_arch.h LuaJIT-riscv_patched/src/lj_arch.h
--- LuaJIT-riscv/src/lj_arch.h	2025-01-23 07:11:19.000000000 +0100
+++ LuaJIT-riscv_patched/src/lj_arch.h	2025-04-15 11:15:32.564000857 +0200
@@ -551,9 +551,9 @@
 #undef LJ_TARGET_MIPS
 #endif
 #elif LJ_TARGET_RISCV64
-#if !defined(__riscv_float_abi_double)
-#error "Only RISC-V 64 double float supported for now"
-#endif
+//#if !defined(__riscv_float_abi_double)
+//#error "Only RISC-V 64 double float supported for now"
+//#endif
 #endif
 #endif
 
